app/context/extra_keyboard/index.tsx

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {createContext, useCallback, useContext, useEffect, useState} from 'react';
+import {Keyboard, Platform} from 'react-native';
+import Animated, {KeyboardState, useAnimatedKeyboard, useAnimatedStyle, useDerivedValue, useSharedValue, withTiming} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import {Screens} from '@constants';
+import {useIsTablet} from '@hooks/device';
+import NavigationStore from '@store/navigation_store';
+import {preventDoubleTap} from '@utils/tap';
+
+import type {AvailableScreens} from '@typings/screens/navigation';
+
+export type ExtraKeyboardContextProps = {
+    isExtraKeyboardVisible: boolean;
+    component: React.ReactElement|null;
+    showExtraKeyboard: (component: React.ReactElement|null) => void;
+    hideExtraKeyboard: () => void;
+    registerTextInputFocus: () => void;
+    registerTextInputBlur: () => void;
+};
+
+// This is based on the size of the tab bar
+const KEYBOARD_OFFSET = -77;
+
+export const ExtraKeyboardContext = createContext<ExtraKeyboardContextProps|undefined>(undefined);
+
+const useOffetForCurrentScreen = (): number => {
+    const [screen, setScreen] = useState<AvailableScreens|undefined>();
+    const [offset, setOffset] = useState(0);
+    const isTablet = useIsTablet();
+    const sub = NavigationStore.getSubject();
+
+    useEffect(() => {
+        const s = sub.subscribe(setScreen);
+
+        return () => s.unsubscribe();
+    }, []);
+
+    useEffect(() => {
+        if (isTablet && screen === Screens.HOME) {
+            setOffset(KEYBOARD_OFFSET);
+        }
+    }, [isTablet, screen]);
+
+    return offset;
+};
+
+export const ExtraKeyboardProvider = (({children}: {children: React.ReactElement|React.ReactElement[]}) => {
+    const [isExtraKeyboardVisible, setExtraKeyboardVisible] = useState(false);
+    const [component, setComponent] = useState<React.ReactElement|null>(null);
+    const [isTextInputFocused, setIsTextInputFocused] = useState(false);
+
+    const showExtraKeyboard = useCallback((newComponent: React.ReactElement|null) => {
+        setExtraKeyboardVisible(true);
+        setComponent(newComponent);
+        if (Keyboard.isVisible()) {
+            Keyboard.dismiss();
+        }
+    }, []);
+
+    const hideExtraKeyboard = useCallback(() => {
+        setExtraKeyboardVisible(false);
+        setComponent(null);
+        if (Keyboard.isVisible()) {
+            Keyboard.dismiss();
+        }
+    }, []);
+
+    const registerTextInputFocus = useCallback(() => {
+        setIsTextInputFocused(true);
+        setTimeout(() => {
+            setExtraKeyboardVisible(false);
+        }, 400);
+    }, []);
```

Comment:
I don't completely understand why the extra keyboard sets visible to false when the input is focused. Is it a mistake, or I am missing something?

Reply:
Since this is one is tricky, I think we should add some comment to explain why we do this.